### PR TITLE
Km ce join user list

### LINKED ISSUE #5: 5. As a user, I want to join an existing shopping list so I can share a shopping list with another person.
Tokens can be shared with other users to allow them to co-manage a given list.
      Joining an existing list consists of the following:
      * User enters an existing token
      * Save the token to `localStorage`
      * Show the user the existing list
      AC:
      * When the user does not already have a token in `localStorage`, on the onboarding/home screen, a simple form is displayed that allows the user to enter a token
      * Entering the token and hitting submit saves the token to `localStorage`, effectively giving them joint control of the list
      * On submit, show an error if the token does not exist
      * If they get an error message, allow them to try again or create a new list
      Notes:
      * As of this story, the welcome/home screen will be feature complete, as reflected in this wireframe: https://github.com/the-collab-lab/smart-shopping-list/blob/main/_resources/wireframes/welcome-screen.png
      * For accessibility reasons, use a `<label>` tag for the "share token" field.
      * You can assume that users will only be sharing a token if the list has actual items on it. So, let's say I create a New Shopping List, but never add any items to it, and haven't saved that particular token to Firestore, we can assume that's a list that doesn't exist.
      * If you would like, you can save the token to Firestore immediately when creating a list, and then a user would be able to share an empty list (see issue 5).

### PR DESCRIPTION
**Pull request by k-monnik**
## Description

This PR allows a user to join an existing list if they input an existing token, and alerts user with an error message if their token does not exist.

## Related Issue

Closes #5. 

## Acceptance Criteria

* When the user does not already have a token in localStorage, on the onboarding/home screen, a simple form is displayed that allows the user to enter a token
* Entering the token and hitting submit saves the token to localStorage, effectively giving them joint control of the list
* On submit, show an error if the token does not exist
* If they get an error message, allow them to try again or create a new list

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|    | :hammer: Refactoring       |
|    | :100: Add tests            |
|    | :link: Update dependencies |
|    | :scroll: Docs              |

### Before

image

### After

image

image

## Testing Steps / QA Criteria

-Go to home page, create a new list to generate a token, and add an item to save the list. Copy token, then delete token from local storage. Paste the token in the new form under "Share Token" and the token should save to local storage, and navigate the user to the associated list.
-If user does not enter a token or enters a nonexistent token and clicks submit, an error alert message should appear.

### CODE REVIEW

**Review by diyanamendoza — APPROVED** [review]
Yay it's working! Looking clean. Thanks team.

**Review by chriskorsak — APPROVED** [review]
Nice job this week Kristen and Cynthia! This is working exactly as described.

File: src/components/AddItem.js [hunk 1]
```diff
@@ -1,10 +1,13 @@
...
 import { useState } from 'react';
 import { db } from '../lib/firebase.js';
+import useFirebaseSnapshot from '../hooks/useFirebaseSnapshot.js';
+import cleanData from '../utils/cleanData.js';
 
 const AddItem = () => {
+  const docs = useFirebaseSnapshot();
   const [itemName, setItemName] = useState('');
-  const [successMessage, setSuccessMessage] = useState('');
+  const [message, setMessage] = useState('');
```
Comment (aprilleone) [inline src/components/AddItem.js:10]: I really like the changes to these names, so much more flexible this way.

**Review by aprilleone — APPROVED** [review]
We went over this code pretty thoroughly on Wednesday. This looks great! Awesome job, team.

**Pull request merged**